blockly-ws-search/test/workspace_search_test.js

```diff
@@ -0,0 +1,283 @@
+      assertNoExtraCurrentStyling(this.blocks, expectedBlocks[0]);
+    });
+
+    test('Match all field and collapsed blocks', async () => {
+      this.workspaceSearch.searchAndHighlight('string', false);
+      var expectedBlocks = [
+          this.testStatementBlockWithInputCollapsed,
+          this.fieldWithOutputConnected,
+          this.fieldWithOutput
+      ];
+      assertEqualsSearchGroup(
+          this.blocks, this.workspaceSearch.blocks_, expectedBlocks);
+      assertNoExtraCurrentStyling(this.blocks, expectedBlocks[0]);
+    });
+
+    test('Preserve current, in results', async () => {
+      this.workspaceSearch.searchAndHighlight('test', false);
+      this.workspaceSearch.setCurrentBlock_(1);
+      // this.testStatementBlock should be current.
+      var expectedBlocks = [
+        this.testBlock,
+        this.testStatementBlock,
+        this.testStatementBlockWithInput,
+        this.testStatementBlockWithInputCollapsed,
+        this.fieldWithOutputConnected,
+        this.fieldWithOutput
+      ];
+      this.workspaceSearch.searchAndHighlight('test', true);
+      assertEqualsSearchGroup(
+          this.blocks, this.workspaceSearch.blocks_, expectedBlocks);
+      assertNoExtraCurrentStyling(this.blocks, expectedBlocks[1]);
+    });
+
+    test('Preserve current, not in results', async () => {
+      this.workspaceSearch.searchAndHighlight('test', false);
+      this.workspaceSearch.setCurrentBlock_(1);
+      // this.testStatementBlock should be current.
+      this.workspaceSearch.searchAndHighlight('string', false);
+      var expectedBlocks = [
+        this.testStatementBlockWithInputCollapsed,
+        this.fieldWithOutputConnected,
+        this.fieldWithOutput
+      ];
+      assertEqualsSearchGroup(
+          this.blocks, this.workspaceSearch.blocks_, expectedBlocks);
+      assertNoExtraCurrentStyling(this.blocks, expectedBlocks[0]);
+    });
+  });
+
+  suite('next()', () => {
+    setup(() => {
+      this.workspaceSearch.blocks_ =
+          [ this.testBlock, this.testStatementBlock ];
+    });
+
+    test('next() with unset current', async () => {
+      this.workspaceSearch.next();
+      var currentIndex = this.workspaceSearch.currentBlockIndex_;
+      assert.equal(currentIndex, 0);
+    });
+
+    test('next() wrap around', async () => {
+      this.workspaceSearch.currentBlockIndex_ = 0;
+      this.workspaceSearch.next();
+      var currentIndex = this.workspaceSearch.currentBlockIndex_;
+      assert.equal(currentIndex, 1);
+      this.workspaceSearch.next();
+      currentIndex = this.workspaceSearch.currentBlockIndex_;
+      assert.equal(currentIndex, 0);
+    });
+  });
+
+  suite('previous()', () => {
+    setup(() => {
+      this.workspaceSearch.blocks_ =
+          [ this.testBlock, this.testStatementBlock ];
+    });
+
+    test('previous() with unset current', async () => {
+      this.workspaceSearch.previous();
+      // No expected current index, but should not throw.
```

Comment:
Should this be a TODO?

Reply:
Not a TODO, meant to be a note to explain why there aren't any asserts